# Proyecto final Data Analitics

Nos piden que analicemos las ventas de una empresa de franquicias de hostelería.  
Nos entregan una serie de ficheros clasificados en dos carpetas una de "maestros" y otra de "ventas". Vamos a ver lo que contiene cada fichero

In [296]:
import pandas as pd
import numpy as np
import os


Abrimos ahora los ficheros para examinar su contenido. Lo haremos en orden alfabético

In [297]:
base = os.path.dirname(os.path.abspath("PROYECTO FINAL")) #Esto hace que nos construya la ruta correctamente en la mayoría de sistemas operativos
ruta_art = os.path.join(base, "data", "maestros", "artículos.xlsx")

df_arti = pd.read_excel(ruta_art, index_col = "id_producto")
df_arti.sample(5)

,familia,categoria,marca,formato,producto,precio_venta,coste
id_producto,,,,,,,
P104,Comida,Bocadillos,NaN,NaN,Queso,4.50,1.26
P108,Comida,Aperitivos,NaN,NaN,Aceitunas,2.00,0.62
P003,Bebida,Refresco,Fanta Limón,Botella,Botella Fanta Limón,2.75,0.69
P106,Comida,Raciones,NaN,NaN,Calamares,8.50,2.53
P007,Bebida,Cerveza,Heineken,Tercio,Tercio Heineken,3.50,1.01


In [298]:
print(f"Nuestro dataframe original de ARTÍCULOS consta de {df_arti.shape[0]} filas y {df_arti.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_arti.size}")
print(f"Del tota de filas, {df_arti.index.nunique()} son valores únicos")

Nuestro dataframe original de ARTÍCULOS consta de 30 filas y 7 columnas
El volumen de datos que contiene nuestro dataframe es: 210
Del tota de filas, 30 son valores únicos


Veamos el tipo de datos que tenemos

In [299]:
df_arti.info()

<class 'pandas.DataFrame'>
Index: 30 entries, P001 to P109
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   familia       30 non-null     str    
 1   categoria     30 non-null     str    
 2   marca         21 non-null     str    
 3   formato       21 non-null     str    
 4   producto      30 non-null     str    
 5   precio_venta  30 non-null     float64
 6   coste         30 non-null     float64
dtypes: float64(2), str(5)
memory usage: 3.1 KB


5 columnas de texto descriptivo de cada producto y 2 númericas con el PVP y el coste. Hay dos columnas (Marca y Formato) que no están completas.

In [300]:
ruta_cp = os.path.join(base, "data", "maestros", "codigos_postales.xlsx")

df_cp = pd.read_excel(ruta_cp,index_col = "Código postal")

df_cp.sample(5)

,Localidad,Provincia,Comunidad autónoma,País,Habitantes
Código postal,,,,,
28801,Alcalá de Henares,Madrid,Comunidad de Madrid,España,195000
8940,Cornellà de Llobregat,Barcelona,Cataluña,España,90000
28109,Alcobendas,Madrid,Comunidad de Madrid,España,120000
2002,Albacete,Albacete,Castilla-La Mancha,España,175000
48991,Getxo,Bizkaia,País Vasco,España,77000


In [301]:
print(f"Nuestro dataframe original de CÓDIGOS POSTALES consta de {df_cp.shape[0]} filas y {df_cp.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_cp.size}")
print(f"Del total de filas, {df_cp.index.nunique()} son valores únicos")

Nuestro dataframe original de CÓDIGOS POSTALES consta de 76 filas y 5 columnas
El volumen de datos que contiene nuestro dataframe es: 380
Del total de filas, 76 son valores únicos


El código postal, aunque es un número, en este caso, es mejor que lo tratemos como un texto, ya que no realizaremos ninguna operación aritmética con este dato

In [302]:
df_cp.index = df_cp.index.astype(str)

In [303]:
df_cp.info()

<class 'pandas.DataFrame'>
Index: 76 entries, 46113 to 28080
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Localidad           76 non-null     str  
 1   Provincia           76 non-null     str  
 2   Comunidad autónoma  76 non-null     str  
 3   País                76 non-null     str  
 4   Habitantes          76 non-null     int64
dtypes: int64(1), str(4)
memory usage: 6.9 KB


Vemos que este fichero contiene un total de 76 registros que corresponden a distintos códigos postales con los datos de ubicación geográfica de cada uno y un campo numérico que indica el número de habitantes, que por el volumen de las cifras, debe corresponder a las localidades

In [304]:
ruta_fq = os.path.join(base, "data", "maestros", "franquiciados.xlsx")

df_fq = pd.read_excel(ruta_fq,index_col = "id_franquiciado")
df_fq.sample(5)

,nombre,sexo,tipo_contrato,perfil,número_locales,multifranquiciado,Unnamed: 7,Unnamed: 8,Unnamed: 9,
id_franquiciado,,,,,,,,,,
F01,Mamen Molinero Velasco,M,Silver,Autoempleo,6,SI,NaN,NaN,NaN,NaN
F19,Daniel Madrid Molinero,H,Basic,Inversor,6,SI,NaN,NaN,NaN,NaN
F17,María González Jorge,M,Silver,Inversor,4,SI,NaN,NaN,NaN,NaN
F06,Mario Campos Cabrales,H,Silver,Inversor,2,SI,NaN,NaN,NaN,NaN
F03,Carlos Herranz Díaz,H,Gold,Inversor,4,SI,NaN,NaN,NaN,NaN


En este fichero de FRANQUICIADOS tenemos un listado de personas y la descripción de su relación con la franquicia. Tambien vemos que hay 3 columnas vacías que eliminaremos a continuación.

In [305]:
df_fq = df_fq.drop(df_fq.columns[[6, 7, 8, 9]], axis=1)
df_fq.info()

<class 'pandas.DataFrame'>
Index: 20 entries, F01 to F20
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype
---  ------             --------------  -----
 0   nombre             20 non-null     str  
 1   sexo               20 non-null     str  
 2   tipo_contrato      20 non-null     str  
 3   perfil             20 non-null     str  
 4   número_locales     20 non-null     int64
 5   multifranquiciado  20 non-null     str  
dtypes: int64(1), str(5)
memory usage: 1.9 KB


In [306]:
print(f"Nuestro dataframe original de FRANQUICIADOS consta de {df_fq.shape[0]} filas y {df_fq.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_fq.size}")
print(f"Del total de filas, {df_fq.index.nunique()} son valores únicos")

Nuestro dataframe original de FRANQUICIADOS consta de 20 filas y 6 columnas
El volumen de datos que contiene nuestro dataframe es: 120
Del total de filas, 20 son valores únicos


Por último, veamos qué contiene el fichero locales

In [307]:
ruta_loc = os.path.join(base, "data", "maestros", "locales.xlsx")

df_loc = pd.read_excel(ruta_loc, index_col = "ID_tienda")
df_loc.sample(5)

,Nombre Tienda,Dirección,Cod Ubicación,Tipo Local,Cód Postal,ID_franquiciado,Alquiler,Fecha Apertura,Fecha fin contrato,Royalties,m2,terraza,salida_humos
ID_tienda,,,,,,,,,,,,,
T053,Bar Spider-Man,"Avenida Libertad, 91",Centro Comercial,Bar,11500,F01,6867,2020-04-05,2030-04-03,0.06,191,NO,SI
T042,Pub Guns N' Roses,"Plaza Rosa, 88",Calle,Pub,46020,F03,805,2024-05-20,2034-05-18,0.05,171,NO,NO
T021,Pub Oasis,"Camino Madrid, 95",Calle,Pub,8042,F11,8152,2020-03-28,2030-03-26,0.04,259,SI,SI
T014,Cervecería Blanco,"Camino Ronda, 96",Calle,Cervecería,28941,F17,2209,2020-05-05,2030-05-03,0.04,256,NO,SI
T061,Cervecería Marfil,"Avenida Cervantes, 98",Centro Comercial,Cervecería,48013,F16,1847,2020-01-24,2030-01-21,0.06,181,NO,SI


Este parece ser el fichero principal, contiene la relación de tiendas, su ubicación (que podremos relacionar con el fichero de CP), el ID_franquiciado (que relacionaremos con el fichero FRANQUICIADOS) y algunos otros datos descriptivos de cada local

In [308]:
df_loc.info()

<class 'pandas.DataFrame'>
Index: 76 entries, T039 to T025
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Nombre Tienda       76 non-null     str           
 1   Dirección           76 non-null     str           
 2   Cod Ubicación       76 non-null     str           
 3   Tipo Local          76 non-null     str           
 4   Cód Postal          76 non-null     int64         
 5   ID_franquiciado     76 non-null     str           
 6   Alquiler            76 non-null     int64         
 7   Fecha Apertura      76 non-null     datetime64[us]
 8   Fecha fin contrato  76 non-null     datetime64[us]
 9   Royalties           76 non-null     float64       
 10  m2                  76 non-null     int64         
 11  terraza             76 non-null     str           
 12  salida_humos        76 non-null     str           
dtypes: datetime64[us](2), float64(1), int64(3), str(7)
memory usage

In [309]:
df_loc["Cód Postal"] = df_loc["Cód Postal"].astype(str)

In [310]:
print(f"Nuestro dataframe original de LOCALES consta de {df_loc.shape[0]} filas y {df_loc.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_loc.size}")
print(f"Del total de filas, {df_loc.index.nunique()} son valores únicos")

Nuestro dataframe original de LOCALES consta de 76 filas y 13 columnas
El volumen de datos que contiene nuestro dataframe es: 988
Del total de filas, 76 son valores únicos


Visto que tenemos 3 ficheros que se complentan, vamos a unirlos en un solo DATAFRAME.

In [311]:
df_locales = df_loc.merge(
    df_fq,
    left_on="ID_franquiciado",
    right_index=True,
    how="left"
)

In [312]:
df_final = df_locales.merge(
    df_cp,
    left_on="Cód Postal",
    right_index=True,
    how="left"
)

In [313]:
df_final.info()

<class 'pandas.DataFrame'>
Index: 76 entries, T039 to T025
Data columns (total 24 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Nombre Tienda       76 non-null     str           
 1   Dirección           76 non-null     str           
 2   Cod Ubicación       76 non-null     str           
 3   Tipo Local          76 non-null     str           
 4   Cód Postal          76 non-null     str           
 5   ID_franquiciado     76 non-null     str           
 6   Alquiler            76 non-null     int64         
 7   Fecha Apertura      76 non-null     datetime64[us]
 8   Fecha fin contrato  76 non-null     datetime64[us]
 9   Royalties           76 non-null     float64       
 10  m2                  76 non-null     int64         
 11  terraza             76 non-null     str           
 12  salida_humos        76 non-null     str           
 13  nombre              76 non-null     str           
 14  sexo   

In [314]:
print(f"Nuestro dataframe FINAL consta de {df_final.shape[0]} filas y {df_final.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_final.size}")
print(f"Del total de filas, {df_final.index.nunique()} son valores únicos")

Nuestro dataframe FINAL consta de 76 filas y 24 columnas
El volumen de datos que contiene nuestro dataframe es: 1824
Del total de filas, 76 son valores únicos


Ahora ya tenemos un solo DataFrame con los datos de todas las tiendas, tanto de su descripción geográfica como de la información de cada franquiciado.

Vamos a ver ahora la carpeta de ventas

In [315]:
ruta_2024_S1 = os.path.join(base, "data", "ventas", "ventas_2024_S1.csv")

df_2024S1 = pd.read_csv(ruta_2024_S1)
df_2024S1.sample(5)

,fecha,id_local,id_producto,cantidad,importe
87723,2024-03-06,T031,P103,3,13.50
155452,2024-05-17,T046,P019,30,105.00
158197,2024-02-17,T048,P104,7,31.50
171903,2024-05-23,T050,P014,16,40.00
103235,2024-02-10,T035,P001,15,41.25


In [316]:
ruta_2024_S2 = os.path.join(base, "data", "ventas", "ventas_2024_S2.csv")

df_2024S2 = pd.read_csv(ruta_2024_S2)
df_2024S2.sample(5)

,fecha,id_local,id_producto,cantidad,importe
192186,2024-11-10,T050,P006,29,72.50
259441,2024-11-10,T067,P012,42,84.00
120012,2024-09-16,T035,P009,18,54.00
90330,2024-08-11,T024,P020,30,60.00
68816,2024-07-22,T019,P015,22,77.00


In [317]:
ruta_2025_S1 = os.path.join(base, "data", "ventas", "ventas_2025_S1.csv")

df_2025S1 = pd.read_csv(ruta_2025_S1)
df_2025S1.sample(5)

,fecha,id_local,id_producto,cantidad,importe
135336,2025-02-07,T034,P018,24,60.00
113004,2025-04-06,T026,P014,24,60.00
110022,2025-06-26,T025,P017,27,81.00
269908,2025-05-10,T061,P104,7,31.50
172152,2025-03-03,T041,P109,7,14.00


In [318]:
ruta_2025_S2 = os.path.join(base, "data", "ventas", "ventas_2025_S2.csv")

df_2025S2 = pd.read_csv(ruta_2025_S2)
df_2025S2.sample(5)

,fecha,id_local,id_producto,cantidad,importe
191905,2025-08-26,T040,P004,5,13.75
291021,2025-07-15,T059,P003,5,13.75
237289,2025-10-27,T048,P021,8,24.00
183297,2025-11-13,T038,P009,43,129.00
355651,2025-12-06,T070,P018,28,70.00


Los 4 ficheros tienen la misma estructura, cada uno contiene las ventas por artículo, día y local de un semestre y año. Así que a continuación, los uniremos en un solo DataFrame que acumule los dos años de ventas.

In [319]:
ventas = [df_2024S1, df_2024S2, df_2025S1, df_2025S2]
df_ventas = pd.concat(ventas)
df_ventas.info()


<class 'pandas.DataFrame'>
Index: 1275181 entries, 0 to 383241
Data columns (total 5 columns):
 #   Column       Non-Null Count    Dtype  
---  ------       --------------    -----  
 0   fecha        1275181 non-null  str    
 1   id_local     1275181 non-null  str    
 2   id_producto  1275181 non-null  str    
 3   cantidad     1275181 non-null  int64  
 4   importe      1275181 non-null  float64
dtypes: float64(1), int64(1), str(3)
memory usage: 80.3 MB


In [320]:
print(f"Nuestro dataframe de VENTAS consta de {df_ventas.shape[0]} filas y {df_ventas.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_ventas.size}")


Nuestro dataframe de VENTAS consta de 1275181 filas y 5 columnas
El volumen de datos que contiene nuestro dataframe es: 6375905


In [321]:
#Convertimos el campo "fecha" para poder operar con él correctamente
df_ventas["fecha"] = pd.to_datetime(df_ventas["fecha"], format="%Y-%m-%d")


Por lo que vimos anteriormente, podemos relacionar esta tabla de ventas con la tabla de artículos por el id_artículo y con la de locales por el id_local. Con esto tendríamos un solo DataFrame con toda la información que nos han proporcionado 

In [322]:
#Primero añadimos a las ventas la información de cada tienda:
df_franquicia = df_ventas.merge(
                df_final,
                left_on = "id_local",
                right_index = True,
                how= "left" 
)

In [323]:
#Y ahora le agramos la descripción de cada artículo:
df_franquicia = df_franquicia.merge(
                df_arti,
                left_on = "id_producto",
                right_index = True,
                how= "left"
)

In [324]:
df_franquicia.info()

<class 'pandas.DataFrame'>
Index: 1275181 entries, 0 to 383241
Data columns (total 36 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   fecha               1275181 non-null  datetime64[us]
 1   id_local            1275181 non-null  str           
 2   id_producto         1275181 non-null  str           
 3   cantidad            1275181 non-null  int64         
 4   importe             1275181 non-null  float64       
 5   Nombre Tienda       1275181 non-null  str           
 6   Dirección           1275181 non-null  str           
 7   Cod Ubicación       1275181 non-null  str           
 8   Tipo Local          1275181 non-null  str           
 9   Cód Postal          1275181 non-null  str           
 10  ID_franquiciado     1275181 non-null  str           
 11  Alquiler            1275181 non-null  int64         
 12  Fecha Apertura      1275181 non-null  datetime64[us]
 13  Fecha fin contrato  1275181 n

In [325]:
print(f"Nuestro dataframe de FINAL DE VENTAS consta de {df_franquicia.shape[0]} filas y {df_franquicia.shape[1]} columnas")
print(f"El volumen de datos que contiene nuestro dataframe es: {df_franquicia.size}")

Nuestro dataframe de FINAL DE VENTAS consta de 1275181 filas y 36 columnas
El volumen de datos que contiene nuestro dataframe es: 45906516


Tenemos un solo DataFrame de más de 1.2Millones de registros con 36 columnas, en total +4.5millones de datos, haremos ahora una visión general de los datos

In [326]:
df_franquicia.describe(percentiles = None).T

,count,mean,min,25%,50%,75%,max,std
fecha,1275181,2025-01-31 22:05:18.454557,2024-01-01 00:00:00,2024-08-11 00:00:00,2025-02-18 00:00:00,2025-08-01 00:00:00,2025-12-31 00:00:00,NaN
cantidad,"1,275,181.00",16.35,1.00,6.00,14.00,23.00,80.00,12.17
importe,"1,275,181.00",48.71,2.00,22.50,42.00,66.50,276.50,33.91
Alquiler,"1,275,181.00","5,578.70",805.00,"3,217.00","5,980.00","7,848.00","9,755.00","2,607.36"
Fecha Apertura,1275181,2022-03-05 15:03:02.794756,2019-01-11 00:00:00,2020-04-05 00:00:00,2022-03-08 00:00:00,2023-11-07 00:00:00,2025-12-25 00:00:00,NaN
Fecha fin contrato,1275181,2032-03-02 15:03:02.794756,2029-01-08 00:00:00,2030-04-03 00:00:00,2032-03-05 00:00:00,2033-11-04 00:00:00,2035-12-23 00:00:00,NaN
Royalties,"1,275,181.00",0.06,0.04,0.05,0.05,0.07,0.07,0.01
m2,"1,275,181.00",266.02,156.00,217.00,268.00,325.00,377.00,65.53
número_locales,"1,275,181.00",5.03,1.00,4.00,5.00,6.00,9.00,2.13
Habitantes,"1,275,181.00","719,618.50","21,000.00","85,000.00","210,000.00","800,000.00","3,200,000.00","1,029,551.40"


Aquí ya podemos extraer algunas conclusiones:  
-El alquiler medio es de unos 5.5K pero hay bastante variación, ya que van desde los 800€ hasta casi 10K.  
-El primer local se abrió en Enero de 2019 y el último a finales de 2025.  
-Los Royalties tienen cierta variación, entre el 4% y el 7%, con un promedio de 5.5%  
-El tamaño estándar de los locales es de unos 260 metros cuadrados  
-Hay franquiciados que tienen un solo local y alguno cuenta hasta con 9 locales  
-Los precios de venta están en un rango de 1.75€ hasta 8.50€


In [327]:
# Veamos las ventas por provincia
tabla_vtas_prov = (
    df_franquicia.groupby("Provincia") #Agrupamos los datos por provincia
    .agg(tot_ventas=("importe", "sum"))#sumamos las ventas
    .round(2) #readondeamos a 2 decimales
    .sort_values("tot_ventas", ascending=False) # y ordenamos de mayor a menor
)
tabla_vtas_prov["%_ventas"] = (
    tabla_vtas_prov["tot_ventas"] / tabla_vtas_prov["tot_ventas"].sum() * 100
).round(1)

tabla_vtas_prov

,tot_ventas,%_ventas
Provincia,,
Madrid,"16,127,632.00",26.00
Barcelona,"7,946,858.25",12.80
Málaga,"5,970,520.00",9.60
Alicante,"4,591,794.25",7.40
Valencia,"4,332,017.75",7.00
Bizkaia,"3,714,102.50",6.00
Murcia,"3,541,213.75",5.70
Sevilla,"3,196,411.75",5.10
Burgos,"2,397,957.25",3.90


Madrid concentra más de la cuarta parte de las ventas, le siguen las provincias de Barcelona y Málaga.  
Veamos ahora como se distribuyen las ventas por familias de productos

In [328]:
tabla_vtas_fami = (
    df_franquicia.groupby("familia") 
    .agg(tot_ventas=("importe", "sum"))
    .round(2) 
    .sort_values("tot_ventas", ascending=False)
)
tabla_vtas_fami["%_ventas"] = (
    tabla_vtas_fami["tot_ventas"] / tabla_vtas_prov["tot_ventas"].sum() * 100
).round(1)

tabla_vtas_fami

,tot_ventas,%_ventas
familia,,
Bebida,"52,120,089.00",83.90
Comida,"9,994,029.50",16.10


Claramente se venden muchas más bebidas que comida. Aquí hay una interesante oportunidad de negocio. Profundicemos en el detalle de las ventas por artículos

In [329]:
tabla_vtas_cate = (
    df_franquicia.groupby("categoria") 
    .agg(tot_ventas=("importe", "sum"))
    .round(2) 
    .sort_values("tot_ventas", ascending=False)
)
tabla_vtas_cate["%_ventas"] = (
    tabla_vtas_cate["tot_ventas"] / tabla_vtas_prov["tot_ventas"].sum() * 100
).round(1)

tabla_vtas_cate

,tot_ventas,%_ventas
categoria,,
Cerveza,"46,335,860.00",74.60
Refresco,"5,784,229.00",9.30
Raciones,"5,363,636.00",8.60
Bocadillos,"3,787,789.50",6.10
Aperitivos,"842,604.00",1.40


El principal negocio de esta empresa es la venta de CERVEZA. Tienen una gran oportunidad de crecimiento si potencian la venta de comida con promociones cruzadas de Bebida + Comida.  
Dada la importancia de esta categoría, vamos a ver cueles son las marcas y formatos que mejor funcionan

In [330]:
df_cervezas = df_franquicia[df_franquicia["categoria"] == "Cerveza"]


In [331]:
matriz_cervezas = df_cervezas.pivot_table(
    values="importe",
    index="marca",
    columns="formato",
    aggfunc="sum",
    fill_value=0
)
matriz_cervezas.style.format(lambda x: f"{x:,.0f}".replace(",", "X").replace(".", ",").replace("X", "."))


formato,Botellín,Doble,Jarra,Tercio
marca,,,,
Alhambra,2.623.852,2.106.870,3.163.335,3.686.036
Estrella Galicia,2.629.188,2.105.766,3.156.996,3.682.480
Heineken,2.625.688,2.109.520,3.155.358,3.699.076
Mahou,2.632.990,2.103.552,3.168.261,3.686.893


Vemos que la distribución por marcas esta bastante proporcionada, no hay ninguna que destaque claramente. Sin embargo, en cuanto a los formatos, vemos que dentro del formato BOTELLA, el tercio predomina sobre el botellín y la Jarra sobre el Doble en el los formatos de BARRIL. Probablemente esto atienda a la diferencia de PVP, para comprobarlo, vamos a generar la misma matriz pero con las CANTIDADES

In [332]:
matriz_cervezas = df_cervezas.pivot_table(
    values="cantidad",
    index="marca",
    columns="formato",
    aggfunc="sum",
    fill_value=0
)
matriz_cervezas.style.format(lambda x: f"{x:,.0f}".replace(",", "X").replace(".", ",").replace("X", "."))

formato,Botellín,Doble,Jarra,Tercio
marca,,,,
Alhambra,1.049.541,1.053.435,1.054.445,1.053.153
Estrella Galicia,1.051.675,1.052.883,1.052.332,1.052.137
Heineken,1.050.275,1.054.760,1.051.786,1.056.879
Mahou,1.053.196,1.051.776,1.056.087,1.053.398


Efectivamente las cantidades son muy similares, por lo que la diferencia de facturación viene dada por un mayor precio de venta

Veamos cuales on los franquiciados que aglutinan más ventas

In [333]:
tabla_vtas_fq = (
    df_franquicia.groupby("nombre") 
    .agg(tot_ventas=("importe", "sum"))
    .round(2) 
    .sort_values("tot_ventas", ascending=False)
)
tabla_vtas_fq["%_ventas"] = (
    tabla_vtas_fq["tot_ventas"] / tabla_vtas_prov["tot_ventas"].sum() * 100
).round(1)

tabla_vtas_fq

,tot_ventas,%_ventas
nombre,,
Sara Posadas García,"6,840,751.50",11.00
José Manuel Canal Prieto,"6,369,940.75",10.30
Daniel Madrid Molinero,"5,760,368.50",9.30
Laura Sánchez Maestro,"5,061,633.50",8.10
Grupo SaDaAm,"4,442,145.50",7.20
Estibaliz Maluenda Azorín,"4,347,293.50",7.00
María González Jorge,"3,931,111.50",6.30
Irene Blanco Latorre,"3,700,035.75",6.00
Mamen Molinero Velasco,"3,388,149.75",5.50


Vamos a ver como se reparten las ventas a lo largo del año

In [338]:
df_franquicia["año"] = df_franquicia["fecha"].dt.year
df_franquicia["mes"] = df_franquicia["fecha"].dt.month

matriz_mensual = df_franquicia.pivot_table(
    values="importe",
    index="mes",
    columns="año",
    aggfunc="sum",
    fill_value=0
)
matriz_mensual

año,2024,2025
mes,,
1,"2,011,751.75","2,634,754.25"
2,"1,929,044.25","2,475,850.50"
3,"2,130,289.50","2,885,647.00"
4,"1,993,891.75","2,750,565.75"
5,"2,166,453.75","2,988,652.25"
6,"2,231,944.50","2,866,115.00"
7,"2,196,036.25","2,992,846.75"
8,"2,357,167.25","3,161,245.25"
9,"2,309,723.50","2,912,859.75"


Una vez que tenemos una visión general de los datos que estamos manejando, pasamos a PowerBi para la parte visualización.   
En este caso, por el caracter académico del proyecto, no utilizaremos las transformaciones realizadas en Python, sino que las reproduciremos en Power Query para aplicar los conocimientos adquiridos. 

In [339]:
df_franquicia.to_csv("ventas_completas.csv", index=False)